lib/c_backend/transform.ml

```diff
@@ -685,6 +685,103 @@ and prepend_expr env ~prepend_stmts ~append_stmts (expr: expr_result) =
 
   result
 
+and do_literal_folding op left right original_expr =
+  let open Expression in
+
+  let do_integer_folding int_op left_int right_int =
+    let new_int =
+      match int_op with
+      (* TODO: i64 support *)
+      | BinaryOp.Minus -> Int32.(-) left_int right_int
```

Comment:
Why not use binary operation style: `Int32.(left_int + right_int)`, or just open the scope of `I32`:
```ocaml
let open Int32 in
match int_op with
| BinaryOp.Minus -> left_int - right_int
```